### PR TITLE
Feature/docker non raspi

### PR DESCRIPTION
**Pull request by thomascrha**
Hi there great project.

I wanted to see if this was something you wanted to include in your project.

This isn't a completed PR - but its more a feel for whether or not you are interested in supporting this kind of functionality in non raspberry pi builds. 

I have a server down stairs and already have some shielded audio cable running to there from my living room. I've ordered a Sonos beam and a record player for my wife for Christmas. My plan is to run this application there inside a docker container.

Currently I haven't tested this with any real Sonos gear - but its running in the container. 

My plan:

- [ ] Add a helper script to generate a docker-compose file to mount the correct USB sound card into the container
- [ ] Get it all working with a sonos device
- [ ] Add to the documentation - basically add a new section in the readme 
- [ ] and probably more things once i get this Sonos speaker

What I wanted to know is if this something that interest you - if not i'll just maintain my own fork.

---
This change is [Reviewable](https://reviewable.io/reviews/wil3/jackcast/18)

### CODE REVIEW

File: jackcast/speakers/sonos.py [hunk 1]
```diff
@@ -57,19 +57,20 @@ def __init__(self, server_port=80):
...
 
-        self.uri ='x-rincon-mp3radio://{}:{}/cast'.format(self.get_ip_addr(), server_port) 
+        self.uri ='x-rincon-mp3radio://{}:{}/cast'.format(self.get_ip_addr(), server_port)
         # The coordinator is the speaker we communicate with. If there is a
-        # group it will forward the request to the other speakers. 
+        # group it will forward the request to the other speakers.
         self.coordinator = None
 
         # Init an active speaker
         devices = soco.discover()
```
Comment (thomascrha) [inline jackcast/speakers/sonos.py:25]: if no sonos devices are detected don't loop through

**Comment by wil3** [comment]
Hey @thomascrha thanks for the contribution. Dockerization would be a great addition and hopefully make it easier for others to adopt Jackcast. Jackcast wasn't intended to run solely on a raspberry pi, during development I was running it off my laptop. For a permanent installation a raspberry pi was just most convenient at the time so its nice to hear you have plans to run on different hardware.

This is the first PR of size for this repo so it'll take me a bit to address it all. I'm going to integrate this repo with Reviewable so it'll be easier to address the PR.  A couple quick comments briefly looking at the changes,
* It would make integrating new code easier and quicker if this PR was broken into smaller PRs. For example one PR for docker support and other PR for refactoring of the core code, etc. Each PR should provide a specific feature so the commit history is well maintained. 
* Your .vscode directory should be added to .gitignore, those files should not be included in the PR
* A requirements.txt shouldn't be necessary, dependencies are in setup.py. If there is need for the requirements.txt that setup.py can't fulfill that should be explained.  

thanks again!
-wil

**Comment by wil3** [comment]
@thomascrha you'll now see a link to Reviewable after your first comment

**Comment by thomascrha** [comment]
Thanks will. Sorry for not getting back to you sonner - I was waiting for the sonos beam to arrive. I'll break down the PR into smaller ones and address your other comments.

**Pull request closed without merging**